### PR TITLE
Feat/INVT-CPC-889_Search_letter_Or_Number_Result_Testings

### PR DESCRIPTION
**Pull request by Matthewtan9**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/CPC-899
What is the ticket about and why are we doing this change.
For search product name and inventory search by fields.
-Added service unit testing
-Added repository testing
-Added integration testing
-Added unit testing
What are the various changes and what other modules do those changes effect.

This can be bullet point or sentence format.
## Before and After UI (Required for UI-impacting PRs)
If this is a change to the UI, include before and after screenshots to show the differences.
If this is a new UI feature, include screenshots to show reviewers what it looks like. 
## Dev notes (Optional)
- Specific technical changes that should be noted
## Linked pull requests (Optional)
- pull request link

### CODE REVIEW

**Review by Eli293 — APPROVED** [review]
Looks fine, make sure to update the branch too.

**Comment by Matthewtan9** [comment]
> Looks fine, make sure to update the branch too.

I reset the other branch because I was committed behind. This is a new branch created from the origin main.  So basically everything should be up to date!

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java [hunk 1]
```diff
@@ -150,6 +150,55 @@ public void testFindProductByProductId() {
                 })
                 .verifyComplete();
     }
+    @Test
+    public void shouldFindProductByInventoryIdAndOneCharNameRegex() {
+
+        String regexPattern = "(?i)^n.*";
```
Comment (AlexeiTimbro) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java:156]: I really like the way you used the regex expression in your testing. It really shows how versatile you are in your craft Matthew keep it up and great work.

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java [hunk 2]
```diff
@@ -150,6 +150,55 @@ public void testFindProductByProductId() {
...
+        // Act & Assert
+        StepVerifier
+                .create(productRepository.findAllProductsByInventoryIdAndProductNameRegex("inventoryId_1", regexPattern))
+                .expectNextMatches(result -> result.getProductId().equals("productId_1"))
+                .verifyComplete();
+    }
+    @Test
+    public void shouldFindProductByInventoryIdAndFullNameRegex() {
+
+        String regexPattern = "(?i)^name.*";
```
Comment (AlexeiTimbro) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java:169]: Here as well

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Inventory/InventoryRepositoryTest.java [hunk 3]
```diff
@@ -104,6 +105,40 @@ public void shouldFindInventoryByName() {
...
+                .verifyComplete();
+    }
+
+    @Test
+    public void shouldFindInventoryByDescriptionRegex() {
+        // Arrange
+        Inventory inventory = buildInventory("inventoryId_5", "SomeName", "Internal", "Medication");
+        inventoryRepository.save(inventory).block();
+
+        String regexPattern = "(?i)^M.*";  // Since we know the description starts with 'M'
```
Comment (AlexeiTimbro) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Inventory/InventoryRepositoryTest.java:129]: Here as well

File: inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java [hunk 4]
```diff
@@ -867,6 +870,166 @@ >>>>>>> d1e75733 (fixing a bit of code because of the quodana requirement)
...
+
+        StepVerifier.create(result)
+                .expectError(NotFoundException.class)
+                .verify();
+    }
+    @Test
+    void getProductsByInventoryIdAndProductName_withValidFields_shouldSucceed(){
+        String inventoryId = "1";
+        String productName = "B";
+        String regexPattern = "(?i)^" + Pattern.quote(productName) + ".*";
```
Comment (AlexeiTimbro) [inline inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java:988]: Here as well

**Review by AlexeiTimbro — APPROVED** [review]
I approve of these changes. Your tests look good and they cover your code very well. Keep it up, twin.

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java (hunk 1 above)
Comment (Matthewtan9) [reply inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java:156]: Wow thank you Alexei Timbro, I am very grateful to hear such words gracefully come out of this comment. I will keep up the work and I hope you do too 😉.

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java (hunk 2 above)
Comment (Matthewtan9) [reply inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Product/ProductRepositoryTest.java:169]: Thank you.

File: inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Inventory/InventoryRepositoryTest.java (hunk 3 above)
Comment (Matthewtan9) [reply inventory-service/src/test/java/com/petclinic/inventoryservice/datalayer/Inventory/InventoryRepositoryTest.java:129]: Thank you.

File: inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java (hunk 4 above)
Comment (Matthewtan9) [reply inventory-service/src/test/java/com/petclinic/inventoryservice/businesslayer/ProductInventoryServiceUnitTest.java:988]: Thank you.

**Pull request merged**